expression/builtin_compare_vec.go

```diff
@@ -590,11 +590,79 @@ func (b *builtinNullEQStringSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinLTIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLTIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	var err error
+	var buf0, buf1 *chunk.Column
+	buf0, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err = b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalString(b.ctx, input, buf1); err != nil {
+		return err
+	}
+
+	result.ResizeInt64(n, false)
+	VecCompareInt(mysql.HasUnsignedFlag(b.args[0].GetType().Flag), mysql.HasUnsignedFlag(b.args[1].GetType().Flag), buf0, buf1, result)
+	result.MergeNulls(buf0, buf1)
+	vecCompareNull(buf0, buf1, result.Int64s())
+	vecResOfLT(result)
+	return nil
+}
+
+func vecResOfLT(result *chunk.Column) {
+	res := result.Int64s()
+	n := len(res)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			res[i] = 0
+			continue
+		}
+		if res[i] < 0 {
+			res[i] = 1
+		} else {
+			res[i] = 0
+		}
+	}
+}
+
+func vecCompareNull(largs, rargs *chunk.Column, res []int64) {
+	n := len(res)
+	for i := 0; i < n; i++ {
+		if largs.IsNull(i) && rargs.IsNull(i) {
+			res[i] = 0
+		} else if largs.IsNull(i) && !rargs.IsNull(i) {
+			res[i] = -1
+		} else if !largs.IsNull(i) && rargs.IsNull(i) {
+			res[i] = 1
+		}
+	}
+}
+
+//VecCompareInt is vectorized CompareInt()
+func VecCompareInt(isUnsigned0, isUnsigned1 bool, largs, rargs, result *chunk.Column) {
+	switch {
+	case isUnsigned0 && isUnsigned1:
+		types.VecComparUU(largs.Uint64s(), rargs.Uint64s(), result.Int64s())
+	case isUnsigned0 && !isUnsigned1:
+		types.VecComparUI(largs.Uint64s(), rargs.Int64s(), result.Int64s())
+	case !isUnsigned0 && isUnsigned1:
+		types.VecComparUI(rargs.Uint64s(), largs.Int64s(), result.Int64s())
```

Comment:
Will this cause a wrong result?
Should we implement VecCompareIU ?

Reply:
I think ‘VecComparUI’ can completely replace ‘ VecCompareIU’, as follows:
![image](https://user-images.githubusercontent.com/13314548/66905584-b2cb1800-f038-11e9-913f-2d20903bee3a.png)

Reply:
I do not think so.
When `types.VecCompareUI(right, left)` returns 1 which means `right > left`, but the caller will think that `left > right`

Reply:
That is indeed wrong： 
5, -5 , VecCompareUI(5, -5) = 1 --correct
-5, 5 , VecCompareUI(5, -5) = 1  --wrong,should return -1
I will fix it. Thank you again!

Reply:
I had added the function " VecComparIU"